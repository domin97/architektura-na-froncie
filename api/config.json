{
  "NAME": "ItCorpo",
  "OPEN_RESOURCES": [
    "/db",
    "/log",
    "/auth",
    "/license",
    "/images/error"
  ],
  "TENANT_ID_HEADER": "TenantIDHeader"
}
