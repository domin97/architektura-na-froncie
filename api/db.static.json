{
  "departments": [
    {
      "id": 1,
      "name": "Management",
      "operatingCosts": 8436
    },
    {
      "id": 2,
      "name": "Sales",
      "operatingCosts": 5836
    },
    {
      "id": 3,
      "name": "Accounting",
      "operatingCosts": 4465
    },
    {
      "id": 4,
      "name": "Human Resources",
      "operatingCosts": 3764
    },
    {
      "id": 5,
      "name": "Marketing",
      "operatingCosts": 7368
    },
    {
      "id": 6,
      "name": "Development",
      "operatingCosts": 9209
    },
    {
      "id": 7,
      "name": "Support",
      "operatingCosts": 2901
    }
  ]
}
