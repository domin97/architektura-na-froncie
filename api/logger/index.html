<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>ANF Logger</title>
  <link type="text/css" rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jsgrid/1.5.3/jsgrid.min.css" />
  <link type="text/css" rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jsgrid/1.5.3/jsgrid-theme.min.css" />
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato">
  <style>
    body {
      font-family: 'Lato', serif;
    }
  </style>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jsgrid/1.5.3/jsgrid.min.js" integrity="sha512-blBYtuTn9yEyWYuKLh8Faml5tT/5YPG0ir9XEABu5YCj7VGr2nb21WPFT9pnP4fcC3y0sSxJR1JqFTfTALGuPQ==" crossorigin="anonymous"></script>
  <script src="app.js"></script>
</head>
<body>
  <h1>system logs</h1>
  <div id="grid"></div>
</body>
</html>
