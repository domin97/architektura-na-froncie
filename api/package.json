{
  "name": "anf-rest-api",
  "description": "Architektura Nowoczesnego Frontendu - API",
  "private": true,
  "author": "devstyle",
  "scripts": {
    "start": "node server",
    "prepare:localdb": "ncp db.json db.local.json",
    "postinstall": "npm run prepare:localdb",
    "generate:mocks": "node lib/mocks.js",
    "generate:types": "node lib/types.js",
    "generate": "npm run generate:mocks && npm run generate:types && npm run prepare:localdb"
  },
  "dependencies": {
    "ajv": "^6.12.6",
    "body-parser": "^1.20.0",
    "chalk": "^4.1.2",
    "connect-pause": "^0.1.0",
    "cors": "^2.8.5",
    "express": "^4.17.3",
    "express-jwt": "^6.1.1",
    "http-server": "^14.1.0",
    "json-schema-faker": "^0.4.7",
    "json-schema-to-typescript": "^10.1.5",
    "json-server": "^0.17.0",
    "jsonwebtoken": "^8.5.1",
    "ncp": "^2.0.0",
    "socket.io": "^4.4.1",
    "uuid": "^8.3.2",
    "ws": "^8.8.0",
    "yargs": "^17.4.1"
  },
  "engines": {
    "node": ">=16.0.0"
  }
}
