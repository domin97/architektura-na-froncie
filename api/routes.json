{
  "/account/transfers*": "/transfers$1",

  "/:resource/:id/count": "/:resource/:id",
  "/:resource/count?:query": "/:resource?:query",
  "/:resource/count": "/:resource"
}
