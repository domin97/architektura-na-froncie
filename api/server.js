const path = require('path')

const jsonServer = require('json-server')
const express = require('express')
const cors = require('cors')
const bodyParser = require('body-parser')

const app = jsonServer.create()
const jsonServerMiddlewares = jsonServer.defaults()
const jsonParser = bodyParser.json()

const DATABASE_FILE = 'db.local.json'
const router = jsonServer.router(DATABASE_FILE)
const countMiddleware = require('./middlewares/count')
const logsMiddleware = require('./middlewares/logs')

router.render = countMiddleware()

const { argv } = require('./lib/cli')
const { logInfo, logConfig } = require('./lib/util')

const pagingMiddleware = require('./middlewares/paging')
const delayingMiddleware = require('./middlewares/delaying')
const failingMiddleware = require('./middlewares/failing')
const errorMiddleware = require('./middlewares/error')

app.use(cors())
app.use(jsonParser, logsMiddleware)
app.use(jsonServer.rewriter(require('./routes.json')))
app.use(jsonServerMiddlewares)

// chaos engineering
const [from, to] = argv.delay.split(':')
app.use(delayingMiddleware({ from, to }))
app.use(pagingMiddleware(50, { excludePatterns: ['logs'] }))
app.use(failingMiddleware(argv.fail, argv.failUrls))

// logger UI
app.use('/logger', express.static(path.join(__dirname, 'logger')))

app.use(require('./middlewares/auth/authorize')(argv.auth))
app.use('/auth', require('./middlewares/auth'))
app.use('/banking', require('./middlewares/banking'))
app.use(require('./middlewares/exams'))
app.use(require('./middlewares/transfers'))
app.use(router)

app.use(errorMiddleware())

app.listen(argv.port, () => {
  logInfo(`JSON Server is running on http://localhost:${argv.p}`)
  logInfo(`Database loaded from file: ${DATABASE_FILE}`)
})

require('./collaborative/register-collaborative-service')(argv.collaborativeService)

require('./currency/register-currency-service')(argv.currencyService) // Socket.IO
// require('./currency/register-currency-service-ws')(argv.currencyService) // "czyste" websockety
