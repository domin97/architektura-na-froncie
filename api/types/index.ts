export * from './transfers'
export * from './log'
