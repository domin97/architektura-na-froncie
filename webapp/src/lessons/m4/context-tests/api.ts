export interface Participant {
  id: number
  name: string
}

export const fetchParticipants = async (): Promise<Participant[]> => {
  // some dummy resource
  const response = await fetch('http://jsonplaceholder.typicode.com/users')
  return response.json()
}
