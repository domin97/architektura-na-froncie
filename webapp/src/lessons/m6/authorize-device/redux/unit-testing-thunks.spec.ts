import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import flushPromises from "flush-promises";

import { TokenInstruction, TokenConfirmation } from "api/token";
import {
  chooseAddDevice,
  chooseAllowOnce,
  submitAddDevice,
  submitDeviceName,
  submitAllowOnce,
  cancelChoice,
  confirmDeviceAdded,
} from "./AuthorizeDeviceProcessRedux";

// import { getTokenInstruction, sendTokenCode } from "../../../../api/token";
jest.mock("../../../../api/token", () => ({
  getTokenInstruction: async (): Promise<TokenInstruction> => {
    return { tokenId: "123", instruction: "Wpisz hasło" };
  },
  sendTokenCode: async (params: TokenConfirmation): Promise<void> => {
    if (params.tokenCode.length != 4) {
      throw new Error("wrong code");
    }
  },
}));

// `redux-mock-store` NIE uruchamia reducera, dlatego NIE będziemy mieli aktualnego stanu
// assertState oczekuje bycia w konkretnym stanie w momencie dispatchowania danego thunka
// i tego w teście mieć nie będzie... dlatego mockujemy assertState, żeby nie rzucało błędami podczas testu
// import { assertState } from "../state-members";
jest.mock("../../state-members");

type Scenario = Function[]
const addDeviceSuccess: Scenario = [
  () => chooseAddDevice(),
  () => submitDeviceName("mój lapcio"),
  () => submitAddDevice("za-długie-hasło"),
  () => submitAddDevice("pswd"),
  () => confirmDeviceAdded(jest.fn()),
];

const allowOnceSuccess: Scenario = [
  () => chooseAllowOnce(),
  () => submitAllowOnce("za-długie-hasło", jest.fn()),
  () => submitAllowOnce("pswd", jest.fn()),
];

const addDeviceCancelThenAllow: Scenario = [
  () => chooseAddDevice(),
  () => submitDeviceName("mój lapcio"),
  () => cancelChoice(),
  () => chooseAllowOnce(),
  () => submitAllowOnce("pswd", jest.fn()),
];

/**
 * 🔥 z racji że mock store nie jest prawdziwym storem, tylko "mockiem", to nie uruchamia
 * reducerów, nie oblicza stanu etc. Dlatego nie przechowuje aktualnego stanu.
 * I kiedy dispatchujemy thunki które ODPYTUJĄ store o aktualny stan, to ten uparcie zwraca initial.
 * Przez co _niektóre_ payloady akcji (te dispatchowane przez thunki odpytujące stan) mają wybrakowany payload.
 *
 * To jest istotne ograniczenie redux-mock-store, gdyby chcieć testować sekwencje akcji
 */
describe("Authorize Device Actions", () => {
  const getMockStore = () => {
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    const initialState = {
      authorizeDevice: {
        type: "CHOOSE_METHOD",
      },
    };
    const store = mockStore(initialState);
    return store;
  }

  it.each<[string, Scenario]>([
    ['addDeviceSuccess', addDeviceSuccess],
    ['allowOnceSuccess', allowOnceSuccess],
    ['addDeviceCancelThenAllow', addDeviceCancelThenAllow],
  ])("%s scenario match actions snapshots", async (scenarioName, scenario) => {
    const store = getMockStore();
    for (const dispatchable of scenario) {
      store.dispatch(dispatchable());
      await flushPromises();
    }
    expect(store.getActions()).toMatchSnapshot();
  })
});
