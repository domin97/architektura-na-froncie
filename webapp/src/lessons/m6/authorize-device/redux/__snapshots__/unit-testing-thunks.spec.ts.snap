// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorize Device Actions addDeviceCancelThenAllow scenario match actions snapshots 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "authorizeDevice/addDeviceForm",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/loading",
  },
  Object {
    "payload": Object {
      "deviceName": "mój lapcio",
      "error": false,
      "instruction": "Wpisz hasło",
      "tokenId": "123",
    },
    "type": "authorizeDevice/addDeviceToken",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/cancelChoice",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/loading",
  },
  Object {
    "payload": Object {
      "error": false,
      "instruction": "Wpisz hasło",
      "tokenId": "123",
    },
    "type": "authorizeDevice/allowOnceToken",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/loading",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/allowOnceSuccess",
  },
]
`;

exports[`Authorize Device Actions addDeviceSuccess scenario match actions snapshots 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "authorizeDevice/addDeviceForm",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/loading",
  },
  Object {
    "payload": Object {
      "deviceName": "mój lapcio",
      "error": false,
      "instruction": "Wpisz hasło",
      "tokenId": "123",
    },
    "type": "authorizeDevice/addDeviceToken",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/loading",
  },
  Object {
    "payload": Object {
      "error": true,
      "type": "CHOOSE_METHOD",
    },
    "type": "authorizeDevice/addDeviceToken",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/loading",
  },
  Object {
    "payload": Object {
      "deviceName": undefined,
    },
    "type": "authorizeDevice/addDeviceConfirmation",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/addDeviceSuccess",
  },
]
`;

exports[`Authorize Device Actions allowOnceSuccess scenario match actions snapshots 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "authorizeDevice/loading",
  },
  Object {
    "payload": Object {
      "error": false,
      "instruction": "Wpisz hasło",
      "tokenId": "123",
    },
    "type": "authorizeDevice/allowOnceToken",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/loading",
  },
  Object {
    "payload": Object {
      "error": true,
      "type": "CHOOSE_METHOD",
    },
    "type": "authorizeDevice/allowOnceToken",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/loading",
  },
  Object {
    "payload": undefined,
    "type": "authorizeDevice/allowOnceSuccess",
  },
]
`;
